src/index.spec.ts

```diff
@@ -5,9 +5,17 @@
  * This product includes software developed at Datadog (https://www.datadoghq.com/).
  * Copyright 2021 Datadog, Inc.
  */
-
+jest.mock("./monitors.ts", () => {
+  return {
+    setMonitors: async (shouldThrow: Boolean) => {
```

Comment:
I'm a bit confused, how does this mock work? Does this mock return in every test that touches monitors.ts? I guess what I mean is, since in the test you set [failOnError to true](https://github.com/DataDog/serverless-plugin-datadog/pull/241/files#diff-18466f9d870e2151722b884f1cb0df6c1530517bde9468ec2a5abbbb8f655210R738) does that automagically invoke this mock?

Reply:
You're right to be confused. Jest, the test system we're using, rewires how nodejs's `module` system and lets us change methods imported by files.

In this case, we're telling Jest to rewire the `ServerlessPlugin`'s dependency on `monitors.ts`, to instead call our mock. Since I don't want it to error every time (but I also don't want to call the datadog backend when we run these tests), I just pass a boolean if a mocked method should throw or not.

Hence, a mock.